src/librustc_save_analysis/span_utils.rs

```diff
@@ -35,14 +35,24 @@ impl<'a> SpanUtils<'a> {
         }
     }
 
-    pub fn make_path_string(&self, path: &FileName) -> String {
-        match *path {
-            FileName::Real(ref path) if !path.is_absolute() =>
-                self.sess.working_dir.0
-                    .join(&path)
-                    .display()
-                    .to_string(),
-            _ => path.to_string(),
+    pub fn make_filename_string(&self, file: &SourceFile) -> String {
+        match &file.name {
+            FileName::Real(path) if !file.name_was_remapped => {
```

Comment:
In practice every file that's going through `SourceMap` will be remapped but here it seemed to be a missing case for not-yet-remapped absolute path so I included it for the sake of completeness.